documents/Libraries/stdlib/sx-glsl/lib/sx_sampling.glsl

```diff
@@ -29,70 +32,102 @@ vec3 sx_normal_from_samples_sobel(float S[9], float _scale)
    return (norm + 1.0) * 0.5;
 };
 
-//
-// Blur using box filter for float samples
-//
-float sx_blur_box_float(float S[9])
-{
-    return (S[0] + S[1] + S[2] + S[3] + S[4] + S[5] + S[6] + S[7] + S[8]) / 9.0;
-}
-
-//
-// Blur using box filter for vec2 samples
-//
-vec2 sx_blur_box_vec2(vec2 S[9])
+// Kernal weights for box filter
+void sx_get_box_weights(inout float W[SX_MAX_SAMPLE_COUNT], int filterSize)
 {
-    return (S[0] + S[1] + S[2] + S[3] + S[4] + S[5] + S[6] + S[7] + S[8]) / 9.0;
+    int filterSize2 = filterSize*filterSize;
+    float value = 1.0 / float(filterSize2);
+    for (int i=0; i<filterSize2; i++)
+    {
+        W[i] = value;
+    }
 }
 
-//
-// Blur using box filter for vec3 samples
-//
-vec3 sx_blur_box_vec3(vec3 S[9])
-{
-    return (S[0] + S[1] + S[2] + S[3] + S[4] + S[5] + S[6] + S[7] + S[8]) / 9.0;
-}
-
-//
-// Blur using box filter for vec4 samples
-//
-vec4 sx_blur_box_vec4(vec4 S[9])
+// Kernel weights for Gaussian filter. Sigma is assumed to be 1.
+void sx_get_gaussian_weights(inout float W[SX_MAX_SAMPLE_COUNT], int filterSize)
 {
-    return (S[0] + S[1] + S[2] + S[3] + S[4] + S[5] + S[6] + S[7] + S[8]) / 9.0;
+    if (filterSize >= 7)
+    {
+        W[0] = 0.000036;  W[1] = 0.000363;  W[2] = 0.001446;  W[3] = 0.002291;  W[4] = 0.001446;  W[5] = 0.000363;  W[6] = 0.000036; 
```

Comment:
Instead of hard-coding all the values, could you use constexpr by chance? Might be more elegant.